tinycrypt/ecc_dsa.c

```diff
@@ -278,9 +287,10 @@ int uECC_verify(const uint8_t *public_key, const uint8_t *message_hash,
 
 	/* Accept only if v == r. */
 	diff = uECC_vli_equal(rx, r);
-	if (diff == 0) {
+	if (diff == 0 && flow_control == (num_bits + 4)) {
```

Comment:
It seems better to check diff for error and return UECC_FAILURE if needed, and after it check flow control and diff again.
In current check if flow control not equal to (num_bits + 4),   UECC_FAILURE will be returned instead of UECC_FAULT_DETECTED.

Reply:
You're right. Updated.